File: Iniciante/1018.rs
use std::io;

fn main() {
  let mut input = String::new();

  io::stdin().read_line(&mut input).unwrap();

  let valor = input.trim().parse::<u32>().unwrap();

  let mut remainder = valor;

  let notas_100 = remainder / 100;
  remainder = remainder % 100;

  let notas_50 = remainder / 50;
  remainder = remainder % 50;

  let notas_20 = remainder / 20;
  remainder = remainder % 20;

  let notas_10 = remainder / 10;
  remainder = remainder % 10;

  let notas_5 = remainder / 5;
  remainder = remainder % 5;

  let notas_2 = remainder / 2;
  remainder = remainder % 2;

  let notas_1 = remainder / 1;

  println!("{}
{} nota(s) de R$ 100,00
{} nota(s) de R$ 50,00
{} nota(s) de R$ 20,00
{} nota(s) de R$ 10,00
{} nota(s) de R$ 5,00
{} nota(s) de R$ 2,00
{} nota(s) de R$ 1,00", valor, notas_100, notas_50, notas_20, notas_10, notas_5, notas_2, notas_1);
}

File: Iniciante/1018/1018.java
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);

    int n = scan.nextInt();

    int resto = n;

    int notas100 = resto / 100;
    resto = resto % 100;

    int notas50 = resto / 50;
    resto = resto % 50;

    int notas20 = resto / 20;
    resto = resto % 20;

    int notas10 = resto / 10;
    resto = resto % 10;

    int notas5 = resto / 5;
    resto = resto % 5;

    int notas2 = resto / 2;
    resto %= 2;

    System.out.println(n);
    System.out.println(notas100 + " nota(s) de R$ 100,00");
    System.out.println(notas50 + " nota(s) de R$ 50,00");
    System.out.println(notas20 + " nota(s) de R$ 20,00");
    System.out.println(notas10 + " nota(s) de R$ 10,00");
    System.out.println(notas5 + " nota(s) de R$ 5,00");
    System.out.println(notas2 + " nota(s) de R$ 2,00");
    System.out.println(resto + " nota(s) de R$ 1,00");
  }
}